chess-AI/model_evaluation.py

```diff
@@ -0,0 +1,185 @@
+
+
+class EasyStockfishPlayer:
+    """"AI" class that is a simple wrapper around the Stockfish engine.
+    """
+    def __init__(self, elo_rating=800):
+        self.stockfish = create_stockfish_wrapper()
+        self.stockfish.set_elo_rating(elo_rating)
+        self.stockfish.depth = 5
+
+    def select_move(self, board):
+        """Sets stockfish state from provided fen and returns best move.
+        """
+        self.stockfish.set_fen_position(board.fen())
+        return chess.Move.from_uci(self.stockfish.get_best_move())
+
+    def __str__(self):
+        return "EasyStockfishPlayer"
+
+
+class HardStockfishPlayer:
+    """"AI" class that is a simple wrapper around the Stockfish engine.
+    """
+    
+    def __init__(self, elo_rating=1800):
+        self.stockfish = create_stockfish_wrapper()
+        self.stockfish.set_elo_rating(elo_rating)
+        self.stockfish.depth = 9
+        self.elo = elo_rating
+
+    def select_move(self, board):
+        """Sets stockfish state from provided fen and returns best move.
+        """
+        self.stockfish.set_fen_position(board.fen())
+        return chess.Move.from_uci(self.stockfish.get_best_move())
+
+    def __str__(self):
+        return "HardStockfishPlayer"
+
+
+class RandomPlayer:
+    """"AI" class that is a simple wrapper around the Stockfish engine, generating random moves.
+    """
+    def select_move(self, board):
+        """Sets stockfish state from provided fen and returns random move.
+        """
+        # if type(move) is not str:
+        #     move = move['Move']
+        return random.choice(list(board.legal_moves))
+
+    def __str__(self):
+        return "RandomStockfishPlayer"
+        
+
+def evaluate_two_players(player1, player2, num_games=20):
+    white = False
+    pointsP1 = 0
+    for _ in range(num_games):
+        white = not(white)
+        board = chess.Board()
+        if white:
+            while True:
+                if isinstance(player1, Knightr0Player):
+                    policy,_ = player1.model(get_cnn_input(board))
+                    board.push_uci(policy_converter.find_best_legal_move(policy, board))
+                else:
+                    board.push(player1.select_move(board))
+                if board.is_game_over(claim_draw=True):
+                    if board.is_checkmate():
+                        pointsP1 += 0.5
+                    pointsP1 += 0.5
+                    break
+                
+                if isinstance(player2, Knightr0Player):
+                    policy,_ = player2.model(get_cnn_input(board))
+                    board.push_uci(policy_converter.find_best_legal_move(policy, board))
+                else:
+                    board.push(player2.select_move(board))
+                if board.is_game_over(claim_draw=True):
+                    if not(board.is_checkmate()):
+                        pointsP1 += 0.5
+                    break        
+        else:
+            while True:
```

Comment:
The following code could probably be extracted out to a method since its mostly duplicated code.